src/wallet/crypter.cpp

```diff
@@ -185,24 +185,22 @@ static bool DecryptSaplingSpendingKey(const CKeyingMaterial& vMasterKey,
     return sk.expsk.full_viewing_key() == extfvk.fvk;
 }
 
+// cs_KeyStore lock must be held by caller
 bool CCryptoKeyStore::SetCrypted()
 {
-    LOCK2(cs_KeyStore, cs_SpendingKeyStore);
     if (fUseCrypto)
         return true;
     if (!(mapKeys.empty() && mapSproutSpendingKeys.empty() && mapSaplingSpendingKeys.empty()))
         return false;
-    fUseCrypto = true;
-    return true;
+    return fUseCrypto = true;
```

Comment:
I suggest reverting this change (just the last two lines). The fact that it's returning the same value that `fUseCrypto` is set to is arguably a coincidence. The return value of `true` means "successful", not "encrypted". Also the original code makes it look less like there might be an `==` vs `=` confusion.

Reply:
Good idea, thanks, done, also rebased onto master, https://github.com/zcash/zcash/pull/3932/commits/e33ebdb8bb84b5cef59561dd40c6be9088a30046.